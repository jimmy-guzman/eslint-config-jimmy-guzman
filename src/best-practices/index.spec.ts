import bestPractices from './index'

describe('bestPractices', () => {
  it('should export bestPractices rules config', () => {
    expect(Object.keys(bestPractices.rules)).toHaveLength(82)
    expect(bestPractices).toMatchSnapshot()
  })
})
