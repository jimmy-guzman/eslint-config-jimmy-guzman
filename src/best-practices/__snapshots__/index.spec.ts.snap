// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bestPractices should export bestPractices rules config 1`] = `
{
  "rules": {
    "accessor-pairs": "warn",
    "array-callback-return": "error",
    "block-scoped-var": "error",
    "class-methods-use-this": [
      "error",
      {
        "exceptMethods": [],
      },
    ],
    "complexity": [
      "error",
      10,
    ],
    "consistent-return": "error",
    "curly": "error",
    "default-case": "error",
    "default-case-last": "error",
    "default-param-last": [
      "error",
    ],
    "dot-notation": "error",
    "eqeqeq": "error",
    "grouped-accessor-pairs": [
      "error",
      "getBeforeSet",
    ],
    "guard-for-in": "error",
    "max-classes-per-file": "error",
    "no-alert": "error",
    "no-caller": "error",
    "no-console": "warn",
    "no-constructor-return": "error",
    "no-div-regex": "error",
    "no-else-return": "error",
    "no-empty-function": [
      "error",
      {
        "allow": [
          "arrowFunctions",
          "functions",
          "methods",
        ],
      },
    ],
    "no-eq-null": "off",
    "no-eval": "error",
    "no-extend-native": "error",
    "no-extra-bind": "error",
    "no-extra-label": "error",
    "no-implicit-coercion": [
      "error",
      {
        "allow": [],
        "boolean": true,
        "number": true,
        "string": true,
      },
    ],
    "no-implicit-globals": "error",
    "no-implied-eval": "error",
    "no-invalid-this": "off",
    "no-iterator": "error",
    "no-label-var": "error",
    "no-labels": "error",
    "no-lone-blocks": "error",
    "no-loop-func": "error",
    "no-magic-numbers": "off",
    "no-multi-str": "error",
    "no-new": "error",
    "no-new-func": "error",
    "no-new-wrappers": "error",
    "no-nonoctal-decimal-escape": "error",
    "no-octal": "error",
    "no-octal-escape": "error",
    "no-param-reassign": "off",
    "no-proto": "error",
    "no-redeclare": "error",
    "no-restricted-globals": [
      "error",
      "event",
      "fdescribe",
    ],
    "no-restricted-properties": "off",
    "no-return-assign": "error",
    "no-return-await": "error",
    "no-script-url": "error",
    "no-self-compare": "error",
    "no-sequences": "error",
    "no-shadow": "error",
    "no-throw-literal": "error",
    "no-undef-init": "error",
    "no-undefined": "off",
    "no-unmodified-loop-condition": "error",
    "no-unused-expressions": [
      "error",
      {
        "allowShortCircuit": false,
        "allowTaggedTemplates": false,
        "allowTernary": false,
      },
    ],
    "no-useless-call": "error",
    "no-useless-computed-key": "error",
    "no-useless-concat": "error",
    "no-useless-constructor": "error",
    "no-useless-rename": "error",
    "no-useless-return": "error",
    "no-var": "error",
    "no-void": "error",
    "no-warning-comments": "warn",
    "prefer-const": "error",
    "prefer-named-capture-group": "off",
    "prefer-promise-reject-errors": [
      "error",
      {
        "allowEmptyReject": true,
      },
    ],
    "prefer-regex-literals": "off",
    "prefer-rest-params": "error",
    "prefer-template": "error",
    "radix": "error",
    "require-await": "error",
    "require-unicode-regexp": "off",
    "strict": "error",
    "symbol-description": "error",
    "vars-on-top": "error",
    "yoda": [
      "error",
      "never",
    ],
  },
}
`;
